R/predict.sgdnet.R

```diff
@@ -181,16 +181,19 @@ predict.sgdnet <- function(object,
                            type = c("link",
                                     "response",
                                     "coefficients",
-                                    "nonzero"),
+                                    "nonzero",
+                                    "class"),
                            exact = FALSE,
                            ...) {
 
   stopifnot(is.logical(exact))
 
   type <- match.arg(type)
 
+  family <- extract_family(object)
+
   if (missing(newx) && type %in% c("link"))
-    stop(paste("new data must be provided for type =", type))
+    stop("new data must be provided for type = ", type)
```

Comment:
This seems inconsistent. 

If `missing(newx)`, shouldn't we just do "in sample" predictions, like base R? I'm assuming we're saving `X` else we wouldn't be able to do in sample predictions for the other `type` values.

Reply:
We're not actually returning x as it is; I just forgot to check for the other pertinent types. Do we really want to store copies of x with each new fit?

Reply:
Maybe we should add a `thin` argument (default TRUE?) indicating whether we save X, y. 

Saving them is definitely more standard for regular `R` functions: `glmnet` is the outlier here, but it seems sane behavior in a big data centic package.

Reply:
That's what I was thinking. I'll save this for a future update and add this as a issue to not have to deal with any other possible merge issues from changes to this branch (since there are now several branches based off of this one). Another possibility is to try to fetch x from the environment using the call and throw a warning while doing so.